autonomica-api/app/services/data_pipeline.py

```
@@ -0,0 +1,828 @@
+    @staticmethod
+    def clean_keyword_data(data: Dict[str, Any]) -> Dict[str, Any]:
+        """Clean and normalize keyword data."""
+        cleaned = data.copy()
+        
+        # Clean keyword field
+        if 'keyword' in cleaned:
+            cleaned['keyword'] = cleaned['keyword'].strip().lower()
+        
+        # Normalize numeric fields
+        numeric_fields = ['search_volume', 'cpc', 'competition', 'keyword_difficulty']
+        for field in numeric_fields:
```

Ruff (0.12.2)

Import `field` from line 16 shadowed by loop variable

(F402)